### ISSUE TITLE
Support for View Creation

### ISSUE THREAD

**Issue by thefatoneinthecorner** [knex/knex#1626]
Hi,

Is there any support for view creation (analogous to createTableIfNotExists and friends)?

Many thanks,

-- Alastair

**Comment by KMayne** [comment]
+1 This would be very useful since Knex acts as an abstraction over different SQL syntaxes between our development (SQLite) and production (Postgres) databases.

**Comment by elhigu** [comment]
Sorry, no such support in knex so far. @KMayne you just need to create own query abstraction with knex.raw depending if it is ran on sqlite or pg.

**Comment by mashaalmemon** [comment]
@elhigu I was looking for the ability to have view creation/maintainence  in migrations as well.

You mention that knex.raw can be used. Can you give a quick example of how to achieve this using knex.raw? I'm assuming you don't mean create the entire view with knex.raw, but just sub in 'create view' vs 'create table' somehow with knex.

We use postgres in our project.

**Comment by elhigu** [comment]
What kind of query you need to have generated?

**Comment by mashaalmemon** [comment]
Something like these (depending on the options we're looking to use at view creation time):

```
CREATE VIEW SELECT a.*, b.* WHERE a.b_id=b.id;
CREATE VIEW SELECT a.*, b.* WHERE a.b_id=b.id WITH LOCAL CHECK OPTION;
CREATE VIEW SELECT a.*, b.* WHERE a.b_id=b.id WITH CASCADED CHECK OPTION;
```

**Comment by elhigu** [comment]
This should work for your cases:

```
knex.schema.raw(`CREATE VIEW ? WITH CASCADED CHECK OPTION`, [
  knex.select('a.*', 'b.*').whereColumn('a.b_id','b.id')
]) 
```

**Comment by gpetrov** [comment]
Any news on views creation in the Knex API? Would be really nice to have an generic API integration instead of database dependent creation commands.

**Comment by kibertoad** [comment]
@gpetrov This functionality is still missing, but any contributions on that front would be dearly appreciated.

**Comment by gpetrov** [comment]
Maybe @rijkvanzanten can help - there is also a great need for query the views schema, next to creating and managing them. All databases, even SQLite have Views support and those are pretty powerful.

**Comment by kibertoad** [comment]
This is now available in 0.95.12-rc4
